azure/datalake/store/transfer.py

```diff
@@ -111,6 +111,35 @@ def __str__(self):
 Chunk = namedtuple('Chunk', 'name state offset expected actual exception')
 
 
+class NoSerLock(object):
+    """
+    Pickle friendly wrapper for threading.Lock (which is non-inheritable
+     so we wrap relevant functions)
+    The main part of this wrapper is that __getstate__ will return an empty
+    tuple so that the actual lock isn't serialized
+    """
+
+    def __init__(self):
+        self._lock = threading.Lock()
+
+    def acquire(self):
```

Comment:
You never seem to call these wrapper methods that re-expose the lock so you might as well just leave them out. And if I'm misreading the code and you actually do then it would be easier to just not hide `self._lock` and call the methods on the underlying lock directly.

Another option, though, is to define `__getattr__()` and delegate all missing attributes to the lock delegate so that all you're doing in this class is defining `__init__()`, `__setstate__()`, and `__getstate__()`.